ui/src/data_explorer/containers/DataExplorer.js

```diff
@@ -73,21 +80,34 @@ const DataExplorer = React.createClass({
   render() {
     const {
       autoRefresh,
+      errorThrownAction,
       handleChooseAutoRefresh,
       timeRange,
       setTimeRange,
       queryConfigs,
       queryConfigActions,
       source,
+      writeLineProtocol,
     } = this.props
-    const {activeQueryIndex} = this.state
+    const {activeQueryIndex, showWriteForm} = this.state
 
     return (
       <div className="data-explorer">
+        {showWriteForm
+          ? <OverlayTechnologies>
+              <WriteDataForm
+                errorThrown={errorThrownAction}
+                onClose={() => this.setState({showWriteForm: false})}
```

Comment:
I had a handler function at one point and ended up removing it and inlining it like this, since all it did was call `setState`